src/build-info/src/lib.rs

```diff
@@ -58,3 +58,17 @@ impl BuildInfo {
             .expect("build version is not valid semver")
     }
 }
+
+pub const fn make_build_info(
+    version: &'static str,
+    sha: &'static str,
+    time: &'static str,
+    target_triple: &'static str,
+) -> BuildInfo {
+    BuildInfo {
+        version,
+        sha,
+        time,
+        target_triple,
+    }
+}
```

Comment:
IMO this function actually makes things less clear! When you construct the struct directly, all these parameters have names; with the function you lose that context. 

Can we make this a `build_info!` macro instead? That expands to the `compile_time_run` stuff? I think that'll have all the desired properties; the `compile_time_run` will execute in the _leaf_, as desired, but you won't need to copy/paste the shell invocations around.